db/warehouse/migrate/20250731134651_add_numeric_priority_scores_to_ce_match_candidates.rb

```diff
@@ -0,0 +1,25 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+class AddNumericPriorityScoresToCeMatchCandidates < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    # Add priority_scores, an array of integers for prioritization
+    add_column :ce_match_candidates, :priority_scores, :integer, array: true
+    add_index :ce_match_candidates, :priority_scores, using: :btree, algorithm: :concurrently
+
+    # Remove existing priority_score single integer column.
+    # This migration doesn't port over existing priority scores, since we can just regenerate candidates
+    safety_assured { remove_column :ce_match_candidates, :priority_score, :integer }
+
+    # Add rank to match rules, so we know the order to apply priority rules
+    add_column :ce_match_rules, :rank, :integer
+    add_index :ce_match_rules, [:owner_type, :owner_id, :rank], unique: true, name: 'index_ce_match_rules_owner_rank_unique', algorithm: :concurrently
```

Comment:
minor: because rules can be either eligibility requirements or priorities, perhaps this field should be called `priority_rank` to make it clearer that it doesn't apply to eligibility requirements. Also the unique constraint can be conditional on rule_type

```suggestion
    add_column :ce_match_rules, :priority_rank, :integer
    add_index :ce_match_rules, [:owner_type, :owner_id, :priority_rank], unique: true, name: 'index_ce_match_rules_owner_rank_unique', where: "rule_type = 'priority_scheme'"